pkg/reconciler/taskrun/resources/taskspec.go

```diff
@@ -237,15 +243,13 @@ func GetStepActionsData(ctx context.Context, taskSpec v1.TaskSpec, taskRun *v1.T
 	for i, step := range taskSpec.Steps {
 		if step.Ref == nil {
 			steps[i] = step
+			updateTaskRunProvenance(taskRun, step.Name, i, nil, stepStatusIndex)
```

Comment:
```suggestion
			updateTaskRunProvenance(taskRun, step.Name, i, nil, stepStatusIndex) // create StepState for inline step with nil provenance
```

add a comment for readability